### PR TITLE
feat: Upgrade k8s API dependencies to v0.28.6

### PR DESCRIPTION
**Pull request by andrewjamesbrown**
### What does this PR do?

Updates the k8s API dependencies to v0.28.6

### Motivation

The datadog-agent mutating webhook is not compatible with k8s-native sidecars, and the k8s API libraries must be on at least v0.28.x to pull in the new API; specifically `restartPolicy: Always` for initContainers.

### Additional Notes

Simple pod reproduction case on k8s 1.29, with datadog mutating webhook enabled:
```
apiVersion: v1
kind: Pod
metadata:
  name: sidecar
  namespace: default
  labels:
    admission.datadoghq.com/enabled: 'true'
spec:
  initContainers:
  - name: init
    image: alpine:latest
    command: [ "/bin/sh", "-c", "sleep 10s" ]
    imagePullPolicy: IfNotPresent
  - name: sidecar
    image: alpine:latest
    command: [ "/bin/sh", "-c", "sleep 120m" ]
    imagePullPolicy: IfNotPresent
    startupProbe:
      exec:
        command:
          - echo
          - startup
      initialDelaySeconds: 5
      periodSeconds: 5
    restartPolicy: Always

  containers:
  - image: alpine:latest
    command: [ "/bin/sh", "-c", "sleep 60m" ]
    imagePullPolicy: IfNotPresent
    name: alpine
  restartPolicy: Always
```

This pod fails with error: `The Pod "sidecar" is invalid: spec.initContainers[1].startupProbe: Forbidden: may not be set for init containers without restartPolicy=Always`

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/kubernetes/apiserver/leaderelection/leaderelection_test.go [hunk 1]
```diff
@@ -257,89 +232,6 @@ func TestSubscribe(t *testing.T) {
 
 }
 
-func TestGetLeaderIPFollower_ConfigMap(t *testing.T) {
```
Comment (andrewjamesbrown) [inline pkg/util/kubernetes/apiserver/leaderelection/leaderelection_test.go:260]: client-go removed support for ConfigMap leases with https://github.com/kubernetes/client-go/commit/27bbe765353fa01ed516b11bde1da82ae6f6c3bc

Comment (AliDatadog) [reply pkg/util/kubernetes/apiserver/leaderelection/leaderelection_test.go:260]: Thanks for the PR. We still need to support using configmaps for leader election even if the official package doesn't as leases were introduced in the version 1.14 of Kubernetes. We imported the solution with this PR: https://github.com/DataDog/datadog-agent/pull/17365/files#diff-4ea44693afeae7147e707e7c16e5840480b78076d00103b9680bec712c7e995e
I think we can keep the tests.

Comment (andrewjamesbrown) [reply pkg/util/kubernetes/apiserver/leaderelection/leaderelection_test.go]: @AliDatadog I think 
3a0141c will do it.  As an aside, is there a plan to deprecate support for k8s <1.14?  At this point it is [5 years old](https://kubernetes.io/blog/2019/03/25/kubernetes-1-14-release-announcement/)...

**Comment by andrewjamesbrown** [comment]
@AliDatadog do you want me to add a releasenote for both DCA and datadog-agent?  Should this be an enhancement or a feature?

**Comment by AliDatadog** [comment]
Hi @andrewjamesbrown 
Thank you for your contribution 🙇 . I opened https://github.com/DataDog/datadog-agent/pull/22500 to reduce the external code. We target a release for 7.52.

**Pull request closed without merging**